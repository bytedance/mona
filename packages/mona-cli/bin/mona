#! /usr/bin/env node

const mona = require('../dist').default;
mona();
