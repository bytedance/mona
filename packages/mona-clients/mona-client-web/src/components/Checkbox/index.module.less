.checkbox {
  -webkit-tap-highlight-color: transparent;
  display: inline-block;

  & .wrapper {
    display: -webkit-inline-flex;
    display: inline-flex;
    -webkit-align-items: center;
    align-items: center;
    vertical-align: middle;
  }

  & .input {
    margin-right: 5px;
    -webkit-appearance: none;
    appearance: none;
    outline: 0;
    background-color: #FFF;
    border: 1px solid #E8E8E8;
    border-radius: 100%;
    width: 22px;
    height: 22px;
    position: relative;
  }

  & .input.checked:before {
    font: normal normal normal 14px/1 "ttui";
    content: "\e601";
    font-size: 28px;
    color: #FFFFFF;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -48%) scale(0.73);
    -webkit-transform: translate(-50%, -48%) scale(0.73);
  }
}