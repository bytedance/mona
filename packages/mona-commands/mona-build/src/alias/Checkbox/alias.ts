import { bubbleEventsAlias, basePropAlias, propAliasMap } from '../prop';
import { CheckboxAlias as Alias } from '../type';

const alias: Alias = {
  ...basePropAlias,
  ...bubbleEventsAlias,
  disabled: propAliasMap.disabled,
  checked: propAliasMap.checked,
  color: propAliasMap.color,
  value: propAliasMap.value,
};

export default alias;
