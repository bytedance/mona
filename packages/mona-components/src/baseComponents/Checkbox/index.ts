import React from 'react';
import createComponent from '../../createComponent';
import { BaseProps } from '../base';

export interface CheckboxProps extends BaseProps {
  value?: string
  disabled?: boolean;
  checked?: boolean
  color?: string
}

export const Checkbox: React.ComponentType<CheckboxProps> = createComponent<CheckboxProps>('checkbox')

