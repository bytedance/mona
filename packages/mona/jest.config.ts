import baseConfig from '../../jestconfig.base';

module.exports = baseConfig;
