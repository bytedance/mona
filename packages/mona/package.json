{
  "name": "@bytedance/mona",
  "version": "0.2.0-alpha.9",
  "description": "Merchant app's developing and building tools",
  "main": "./dist/index",
  "scripts": {
    "build": "yarn clean && ttsc -b",
    "start": "ttsc -b -w",
    "clean": "rm -rf dist",
    "test": "jest"
  },
  "files": [
    "dist"
  ],
  "author": "zhangxiaowei.chris@bytedance.com",
  "license": "MIT",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "devDependencies": {
    "copy-webpack-plugin": "^10.2.0",
    "react": "^17.0.2",
    "webpack": "^5.58.2",
    "webpack-chain": "^6.5.1"
  },
  "peerDependencies": {
    "react": "^17.0.2"
  },
  "gitHead": "cc5aa5283af595fd7d8a7b4921f54e2797504edd"
}
