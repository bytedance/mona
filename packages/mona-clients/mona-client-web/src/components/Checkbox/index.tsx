import React, { useState, useEffect, useContext, useRef } from 'react';
import { CheckboxProps } from '@bytedance/mona';
import cs from 'classnames';
import { CheckboxGroupContext } from '../CheckboxGroup';
import styles from './index.module.less';
import { useHandlers } from '../hooks';
import { formatMouseEvent } from '../utils';

const Checkbox: React.FC<CheckboxProps> = (props) => {
  const {
    value,
    disabled,
    checked,
    color = '#F85959',
    children,
    id,
    ...restProps
  } = props;

  const group = useContext(CheckboxGroupContext);
  const orderRef = useRef(-1);

  const { handleClassName, onClick, ...handlerProps } = useHandlers(restProps);

  const [isChecked, setIsChecked] = useState(checked);

  const handleClick = (e: React.MouseEvent) => {
    const newIsChecked = !isChecked;
    setIsChecked(newIsChecked);
    if (orderRef.current >= 0) {
      group?.toggleChecked(orderRef.current, newIsChecked, formatMouseEvent({ event: e, type: 'change' }));
    }
    onClick(e);
  }
  
  // don't change order of useEffects, the order is important
  useEffect(() => {
    setIsChecked(checked);
    if (orderRef.current >= 0) {
      group?.toggleChecked(orderRef.current, checked);
    }
  }, [checked])

  useEffect(() => {
    orderRef.current = group ? group?.initValue(value, checked) : -1;
  }, [])

  useEffect(() => {
    if (orderRef.current >= 0) {
      group?.changeValue(orderRef.current, value);
    }
  }, [value])

  return (
    <div
      className={handleClassName(styles.checkbox)}
      {...handlerProps}
    >
      <div className={styles.wrapper}>
        <div className={cs(styles.input, {[styles.checked]: isChecked})} style={{ borderColor: color, backgroundColor: isChecked ? color: '' }}>
          <input value={value} id={id} onClick={handleClick} type="checkbox" className={styles.inner}></input>
        </div>
        {children}
      </div>
    </div>
  )
}

export default Checkbox;
