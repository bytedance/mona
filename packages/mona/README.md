# @bytedance/mona
对外暴露提供构建时API
## 配置创建
### createProjectConfig
创建项目配置，`mona.config.ts`中使用
#### 例子
```js
// mona.config.ts
import { createProjectConfig } from '@bytedance/mona';
export default createProjectConfig({
  projectName: 'demo',
  input: './src/app.tsx',
  output: 'dist'
})
```

#### 参数说明
|  参数   | 说明  | 是否必填 | 类型 | 默认值 |
|  ----  | ----  | ---- | ---- | ---- |
| projectName  | 应用/插件名称 | 是 | string | - |
| input  | 入口文件路径 | 否 | string | - |
| output  | 打包后的文件目录路径 | 否 | string | - |
| raw  | 定义函数来修改webpack配置 | 否 | (options: any) => any | - |
| dev  | 用来修改本地服务配置 | 否 | string | { port: 9999 } |

### createAppConfig
创建应用配置，`app.config.ts`中使用
#### 例子
```js
// app.config.ts
import { createAppConfig } from '@bytedance/mona';
export default createAppConfig({
  pages: [
    'pages/Home/index',
    'pages/Info/index',
    'pages/List/index'
  ]
})
```
#### 参数说明
|  参数   | 说明  | 是否必填 | 类型 | 默认值 |
|  ----  | ----  | ---- | ---- | ---- |
| pages  | 页面路径 | 是 | string[] | - |

默认`pages`的第一项为首页
### createPageConfig
创建页面配置，需在每个页面目录的`page.config.ts`中使用

#### 例子
```js
// page.config.ts
import { createPageConfig } from '@bytedance/mona';
export default createPageConfig({
  navigationBarTitleText: 'page title'
})
```
#### 参数说明
|  参数   | 说明  | 是否必填 | 类型 | 默认值 |
|  ----  | ----  | ---- | ---- | ---- |
| navigationBarTitleText  | 当前页面标题 | 否 | string | - |
