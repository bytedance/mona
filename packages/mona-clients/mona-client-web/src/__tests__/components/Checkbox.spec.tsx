import React from 'react';
import { mount } from 'enzyme';
import Checkbox from '../../components/Checkbox';
import mountTest from '../../../../../tests/shared/mountTest';
import handlerTest from '../../../../../tests/shared/handlerTest';
import { CheckboxProps } from '@bytedance/mona';

describe('web component: Checkbox', () => {
  mountTest(Checkbox)
  handlerTest({ Component: Checkbox, clickTag: 'input' });

  it('should render correctly', () => {
    expect(mount(<Checkbox>Hello world</Checkbox>).render()).toMatchSnapshot();
  })
  
  it('should have right properties', () => {
    const props: CheckboxProps = {
      value: 'apple',
      checked: true,
      disabled: true,
      color: 'red',
      className: 'class-name',
      style: { color: 'red' },
    }
    const wrapper = mount(<Checkbox {...props}>hello</Checkbox>);
    expect(wrapper.render()).toMatchSnapshot();
  })
})
