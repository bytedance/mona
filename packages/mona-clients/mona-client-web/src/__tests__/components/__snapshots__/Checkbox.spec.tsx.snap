// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web component: Checkbox should have right properties 1`] = `
LoadedCheerio {
  "0": <div
    class="checkbox class-name"
    style="color: red;"
  >
    <div
      class="wrapper"
    >
      <div
        class="input checked"
        style="border-color: red; background-color: red;"
      >
        <input
          class="inner"
          type="checkbox"
          value="apple"
        />
      </div>
      hello
    </div>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`web component: Checkbox should render correctly 1`] = `
LoadedCheerio {
  "0": <div
    class="checkbox"
  >
    <div
      class="wrapper"
    >
      <div
        class="input "
        style="border-color: #f85959;"
      >
        <input
          class="inner"
          type="checkbox"
          value=""
        />
      </div>
      Hello world
    </div>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
